stdlib/src/collections/list.mojo

```diff
@@ -493,16 +493,67 @@ struct List[T: CollectionElement, hint_trivial_type: Bool = False](
         self.capacity = new_capacity
 
     fn append(mut self, owned value: T):
-        """Appends a value to this list.
+        """Appends a value to this list. If there is no capacity left, resizes
+        to twice the current capacity. Except for 0 capacity where it sets 1.
```

Comment:
**Suggestion** I think it might be more readable at-a-glance if we split this description across multiple lines so that the single-line summary stays short and sweet. The allocation strategy could instead be the next paragraph in the docs 🙂 

I.e.

```mojo
"""Appends a value to this list.

If there is no capacity left, resizes to twice the current
capacity. Except for 0 capacity where it sets 1.

...
```

Reply:
will do, but I'd rather we make it the norm to send it to it's own section:
```
Notes:
    ...
```